images/vscode-ssh-server/README.md

```diff
@@ -0,0 +1,94 @@
+# Marinated Concrete's VSCode-compatible ssh server
+This runs a little ssh server that's compatible with VSCode's remote SSH feature. Unfortunately, [the LinuxServer one](https://docs.linuxserver.io/images/docker-openssh-server/) cannot be used because it is based on Alpine so it lacks glibc, a [known limitation](https://code.visualstudio.com/docs/remote/ssh#_remote-ssh-limitations).
+
+It's also less flexible/more opinionated. The intention is to run it as a sidecar to an existing application you have, such as esphome, HomeAssistant, or zwavejs.
+
+## Configuration:
+In a new container alongside your application's container:
+- Set the environment variable `AUTHORIZED_KEYS_URL` to a path containing your authorized keys. For instance, `https://github.com/$GH_USERNAME.keys`.
+- Mount a small empty dir in `/config` - this will container your server's ssh host key, which must be persisted
+- Mount the volume you want accessible over ssh in `/data`. When you ssh in, this is the directory where you'll start.
+
+## Examples
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: esphome
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: esphome
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: esphome
+    spec:
+      containers:
+      - name: esphome
+        env:
+        - name: ESPHOME_DASHBOARD_USE_PING
+            value: "true"
+        image: ghcr.io/esphome/esphome
+        volumeMounts:
+            - mountPath: "/config"
+              name: esphome-config
+        # ... the rest of the esp config
+        # Now add the sidecar:
+      - name: ssh-server
+        env:
+        - name: AUTHORIZED_KEYS_URL
+          # Pick swap in your github username
+          value: "https://github.com/marinatedconcrete.keys"
+        image: ghcr.io/marinatedconcrete/config/vscode-ssh-server
+        ports:
+        - containerPort: 2222
+        volumeMounts:
+        - name: config
+          mountPath: /config
+        - name: data
+          mountPath: /data
+      volumes:
+      - name: data
+        persistentVolumeClaim:
+          claimName: esphome-configs
```

Comment:
IMO, this should be locked down with a much tighter `securityContext`.  Something like this would make sense:
```
          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            runAsUser: 65534
            seccompProfile:
              type: RuntimeDefault
```
`runAsUser` should be whatever UID `dev` is.

It might also make sense to make an `esphome` kustomize component here and just point the docs to that since it'd then be a real, tested usage.  I've had one since November 2023, but it looks like you've deviated from things I setup up with that so I couldn't use this as-is.

Reply:
Do you mind if I publish this first and then come back to update the docs? I am less confident in the security settings so I don't want to document something that is wrong/won't work.

Also not sure if I should just put the security context settings on the pod instead - that might help with `fsGroup` if that's needed? I know esphome can't run as non-root, so there's some scoping to be done to truly figure things out. I'll set an explicit UID in the Dockerfile though.